posts/h5ad_conversion/index.qmd

```diff
@@ -0,0 +1,612 @@
+  
+  # update global offset for next slide
+  global_y_offset <<- global_y_offset + y_span*(1+spacerFactor)
+  
+  return(eachCoord)
+})
+spatial_coords <- do.call(rbind, spatial_coords)  
+
+# use the study-level Seurat object for downstream
+seu.obj1 <- seu.obj
+
+# prefix for file name
+slideName <- "StudyLevel" 
+
+```
+
+Let's visualize the coordinates of all cells after slide arrangement. 
+```{r vizSlides, message=FALSE, eval = FALSE}
+# add in cell type for color
+plotData <- cbind(seu.obj1[["nb_clus_final"]], 
+                  spatial_coords[colnames(seu.obj1), ])
+
+ggplot2::ggplot(plotData, 
+                # flip xy to be consistent with earlier plots
+                aes(x = y, y = x, color = as.factor(nb_clus_final))) +
+  ggplot2::geom_point(size = 0.1)+
+  ggplot2::coord_fixed()+
+  Seurat::NoLegend()+
+  Seurat::DarkTheme()
+```
+![](figures/Fig-mBrain-S1S2-cellType-slidesArranged.png)
+
+
+## Clean up to keep only necessary data {#sec-cleanUp}
+
+Next, we would further clean up the `Seurat` object (single-slide object from [Case 1 in section -@sec-perSlide] or full-study-level object from [Case 2 in section -@sec-perStudy]) by trimming it down to contain only the data of interested. 
+
+Typically, one would keep the raw data counts from `RNA` assay (this example dataset uses `Nanostring` as assay name for `RNA`), cell embedding for `umap` (standard AtoMx exported object uses `approximateumap` as name for `umap`). We would also store the spatial coordinates of each cell as the cell embedding in a dimension reduction object called `spatial`.
+```{r dietSeurat, message=FALSE, eval = FALSE}
+# clean up seurat object to only necessary data 
+seu.obj2 <- Seurat::DietSeurat(
+  seu.obj1, 
+  
+  # subset of assays to keep, standard AtoMx object uses "RNA" assay
+  # of note, AtoMx stores normalized RNA counts in separate "RNA_normalized" assay
+  assays = "Nanostring",
+  
+  # keep raw counts stored in "counts" layer
+  # use "data" or "scale.data" if prefer to keep normalized counts before or after scaling
+  layers = "counts", 
+  
+  # dimension reduction to keep, standard AtoMx object uses "approximateumap"
+  dimreducs = "umap")
+
+# clear the `images` slot
+allImgs <- names(seu.obj1@images)
+for (img in allImgs){
+  seu.obj2[[img]] <- NULL
+}
+
+# add in spatial coordinates for current slide or study as a new dimension reduction
+colnames(spatial_coords) <- paste0("SPATIAL_", seq_len(ncol(spatial_coords)))
+seu.obj2[["spatial"]] <- Seurat::CreateDimReducObject(
+  embeddings = spatial_coords, 
+  key = "SPATIAL_", 
+  # standard AtoMx object use "RNA" assay
+  assay = "Nanostring")
+
+```
+
+## Converting to `h5ad` format via `h5Seurat`
+
+Lastly, we would export the cleaned `Seurat` object into `h5Seurat` format and then further convert it into `h5ad` format using `SeuratDisk::Convert` function.
+```{r h5Seurat, message=FALSE, eval = FALSE}
+# export as "h5Seurat" object
+SeuratDisk::SaveH5Seurat(seu.obj2, 
```

Comment:
Show loading of seuratDisk library if needed

Reply:
In most case, I would prefer not to load libraries to my current environment but calling them via `::`. This way you are not confused by which function comes from which and keep the work environment clean. 
One should have `SeuratDisk` installed it if they have `Seurat` installed.